scipy/spatial/_qhull.pyx

```diff
@@ -2870,6 +2870,15 @@ class HalfspaceIntersection(_QhullUser):
         of halfspaces is also not possible after `close` has been called.
 
         """
+        # We check for non-feasibility of incremental additions
+        # in a manner similar to `qh_sethalfspace`
+        dims = halfspaces.shape[1] - 1
+        for halfspace in halfspaces:
+            dist = halfspace[-1]
+            dist += np.dot(halfspace[:dims], self.interior_point[:dims])
+            if dist > 0:
+                msg = f"feasible point is not clearly inside halfspace: {halfspace}"
+                raise QhullError(msg)
```

Comment:
@RichardBowenGM what do you think of this shim? I suppose we could try to vectorize it (for performance) and potentially report the full set of violating half spaces as well, rather than just the first one encountered, though this may be half decent as a start.

Also, I'm happy to add you as a commit co-author if you tell me what to put per: https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors